### PR TITLE
add improvements and refactorings from other projects

### PR DESCRIPTION
**Pull request by mv-shallow**

### CODE REVIEW

File: src/main/java/com/jupitertools/datasetroll/expect/MatchDataSets.java [hunk 1]
```diff
@@ -26,57 +23,62 @@
...
+                                               List<Map<String, Object>> pattern) {
+        if (matched.size() == pattern.size()) {
+            return;
+        }
+
+        throw new AssertionError(String.format("expected %d but found %d - %s entities",
+                                               pattern.size(), matched.size(), documentName));
+    }
+
+    private void assertSetsOfDocumentsAreIdentical(Set<String> matched, Set<String> pattern) {
```
Comment (mv-shallow) [inline src/main/java/com/jupitertools/datasetroll/expect/MatchDataSets.java:71]: На основе https://github.com/jupiter-tools/spring-test-mongo/blob/0e58b4bd97d21c980bebe06e14e86f21f0e415e0/src/main/java/com/jupiter/tools/spring/test/mongo/internal/expect/MatchDataSets.java#L73

Без сторонней библиотеки

File: src/main/java/com/jupitertools/datasetroll/expect/match/smart/date/TimeDescription.java [hunk 2]
```diff
@@ -83,7 +82,7 @@ private TimeUnit parseTimeUnit(Matcher matcher) {
 
 
     private int parseCount(Matcher matcher) {
-        return Integer.valueOf(matcher.group(4));
+        return Integer.parseInt(matcher.group(4));
```
Comment (mv-shallow) [inline src/main/java/com/jupitertools/datasetroll/expect/match/smart/date/TimeDescription.java:85]: Рефакторинг от IDEA

File: src/main/java/com/jupitertools/datasetroll/expect/match/smart/regexp/MatchRegExp.java [hunk 3]
```diff
@@ -21,7 +20,7 @@ public boolean match(Object originValue, Object comparableValue) {
         String cmpValue = (String) comparableValue;
         cmpValue = cmpValue.replaceFirst("regex:", "").trim();
         Pattern pattern = Pattern.compile(cmpValue);
-        Matcher matcher = pattern.matcher((String) originValue);
+        Matcher matcher = pattern.matcher(originValue.toString());
```
Comment (mv-shallow) [inline src/main/java/com/jupitertools/datasetroll/expect/match/smart/regexp/MatchRegExp.java:23]: Из https://github.com/jupiter-tools/spring-test-mongo/pull/71/files#diff-09e16b6f2e1a77535a10ec330ba6976bf840c3bce5f6d58eb34f652ff9bdf2c6

File: src/main/java/com/jupitertools/datasetroll/exportdata/scanner/CaseInsensitiveMap.java [hunk 4]
```diff
@@ -23,4 +21,9 @@ public Class<?> put(String key, Class<?> value) {
     public Class<?> get(Object key) {
         return super.get(((String) key).toLowerCase());
     }
+
+    @Override
+    public boolean containsKey(Object key) {
```
Comment (mv-shallow) [inline src/main/java/com/jupitertools/datasetroll/exportdata/scanner/CaseInsensitiveMap.java:26]: Из https://github.com/jupiter-tools/spring-test-mongo/pull/71/files#diff-c47b937c5f16a386b77173d35f3df5c2d1e44acff813f51a0159e86979461878

**Comment by antkorwin** [comment]
@mv-shallow thanks a lot for this part of refactoring!

**Pull request merged**